### DISCUSSION TITLE
[HELP] Guide for using Influx db with Speedtest-Tracker

### DISCUSSION THREAD

**Discussion by rhcp011235** [alexjustesen/speedtest-tracker#371] [Q&A]
Hello.

I have this running on my Unraid Instance and it has been great, I finally got around to getting it hooked into InfluxDB2 about a week or so ago.

I saw some people with grafana library in InfluxDB getting a Nice graph working and I am not too keen on influxDB.

I know more Oracle/DB2/Mysql type stuff.

Any help on getting a nice dashboard in Influx would be helpful to everyone :)

Thank you

**Comment by alexdelprete** [comment]
You don't really expect a guide on using Grafana, right? :)

But you're a really lucky person, I already prepared the dashboard for **influxdb v2**: https://github.com/alexjustesen/speedtest-tracker-docs/issues/4#issuecomment-1323035795

If you confirm the intructions to import the dash are ok, @alexjustesen will publish it in the official docs of the project.

Let us know.

![image](https://user-images.githubusercontent.com/7027842/201738108-c9e00772-99ed-4f37-b033-998ca435ebf6.png)

**Comment by rhcp011235** [comment]
> You don't really expect a guide on using Grafana, right? :)
> 
> But you're a really lucky person, I already prepared the dashboard for **influxdb v2**: [alexjustesen/speedtest-tracker-docs#4 (comment)](https://github.com/alexjustesen/speedtest-tracker-docs/issues/4#issuecomment-1323035795)
> 
> If you confirm the intructions to import the dash are ok, @alexjustesen will publish it in the official docs of the project.
> 
> Let us know.
> 
> ![image](https://user-images.githubusercontent.com/7027842/201738108-c9e00772-99ed-4f37-b033-998ca435ebf6.png)

This requires you to install Grafana eh? Is there anyway to simply use Influx? without adding more to it? I am very new to all this and its kind of complex :)

**Comment by rhcp011235** [comment]
not sure why it closed it.

**Comment by alexdelprete** [comment]
You can reopen it...can't you?



> This requires you to install Grafana eh? Is there anyway to simply use Influx? without adding more to it? I am very new to all this and its kind of complex :)

How do you want to display the graphs? Grafana does that. You need a tool.

Maybe in the future @alexjustesen can implement some graphs in the dashboard, but since there's influx, I wanted to see data immediately.

Grafana is simple...check some guides on youtube...once you have it installed, just follow the procedure above to import my dashboard.

**Comment by rhcp011235** [comment]
@alexdelprete I will try, I am very very new to all of this. For years ive been a DBA (But Cli based stuff that IMHO seems simple)

All of these GUI/Point and click things seem to be confusing to me, because it basically just does whatever vs having you set it up. so you really dont learn much. 

If you feel like helping me :) I have Skype and such. :)

**Comment by alexjustesen** [comment]
@rhcp011235 thanks for using Speedtest Tracker! I too used to be a dba, then did a bit of analytics work but during that was always a web developer so this project feels like home.

**Comment by rhcp011235** [comment]
@alexjustesen Thank you! I bet it does for you, I never got into any web dev nor dealt with graphics. Ive always been behind the scenes doing actual DB work and Data mining and Data warehouse type stuff. so this is all like Korean to me :)

**Comment by alexdelprete** [comment]
> For years ive been a DBA

After a couple of years of development, for years I've done system & network management, then project management, then presales and Sales. A little bit of everything. :)



> All of these GUI/Point and click things seem to be confusing to me, because it basically just does whatever vs having you set it up

Oh that is not correct. I'm playing with Grafana since 3-4 months...you really need to understand databases, queries, etc. You would be very good with Grafana. Usually I suck with UI/GUI stuff...but Grafana is pretty good on that side once you understand data, where it is, and what you want to correlate and present. Install it and play with it (with my dashboard for example), with your kind of experience, I'm sure you'll like it. You start with something and customize it, to learn how to use it. Try it.

**Comment by rhcp011235** [comment]
Got it working :) Thank you @alexjustesen @alexdelprete

**Comment by rhcp011235** [comment]
closed

**Comment by alexdelprete** [comment]
> Got it working :) Thank you @alexjustesen @alexdelprete

I told you it was easier than you thought. :)

So the steps of the procedure work? Can we publish them in the official docs?

**Comment by rhcp011235** [comment]
> > Got it working :) Thank you @alexjustesen @alexdelprete
> 
> I told you it was easier than you thought. :)
> 
> So the steps of the procedure work? Can we publish them in the official docs?

Semi.

I had to modify the queries because I named my bucket Speedtest-tracker. Other than that? It was very simple

**Comment by alexdelprete** [comment]
> I had to modify the queries because I named my bucket Speedtest-tracker. Other than that? It was very simple

You can rename a bucket in InfluxDB v2. I will add that to the procedure in case the user has already created the bucket.

![image](https://user-images.githubusercontent.com/7027842/204932889-158b9eef-0e4d-4079-aa2b-5189f0ddef49.png)

**Comment by PovilasID** [comment]
Hey,
I made a selector variable for buckets and made all panels listen for it. Now you do not need to rename buckets in InfluxDB. 
![image](https://user-images.githubusercontent.com/396243/214298929-519b01a2-2353-4619-8b00-384841f5508f.png)

Here is Gist for it https://gist.github.com/PovilasID/113c5d2ef6fc9fbddcd19864466399a6
I suggest adding Grafana dashboard's export in the repo so it can be versioned

**Comment by alexdelprete** [comment]
> I made a selector variable for buckets and made all panels listen for it.

Thanks. I applied the variable to the panels and uploade v2 to grafana dashboards.
There was also a bug when multiple servers where used, I modified the queries and now should be good also in that case.

![image](https://user-images.githubusercontent.com/7027842/214446087-7ea553c1-896b-48a0-8b1f-d8ffcf049b80.png)

If somebody could test the import (to make sure latest revision is used) and report back it would be very helpful. Thanks in advance.

**Comment by PovilasID** [comment]
> If somebody could test the import

I tested dash id `17460` looks good 👍

**Comment by alexdelprete** [comment]
Thanks a lot for the help. Do you have multiple speedtest servers configured? Because the dash wasn't working properly in those cases. I fixed it.

**Comment by PovilasID** [comment]
> Thanks a lot for the help. Do you have multiple speedtest servers configured? Because the dash wasn't working properly in those cases. I fixed it.

Not entire sure... a) Not all testing servers get passed to my influxdb instance because of #301 so I have only a couple of records. b) They are from different servers and all the charts look populated... so it looks like it is fixed but I can not validate 100%

**Comment by PovilasID** [comment]
I updated the latest version with some bar charts:
![image](https://user-images.githubusercontent.com/396243/215275478-5e81b083-d1e1-46be-9625-c667fd1377ae.png)

https://gist.github.com/PovilasID/113c5d2ef6fc9fbddcd19864466399a6

**Reply by PovilasID** [reply]
I also added a row with 4 charts displaying per server dynamics of download upload speeds and ping and jitter dynamics. I also enabled shared crosshair to enabled shared crosshair. 
![image](https://user-images.githubusercontent.com/396243/215433777-746f1b43-2689-4de3-8aaf-574a300826db.png)

**Reply by PovilasID** [reply]
Also added a server selector for per server results comparison

**Comment by alexdelprete** [comment]
Where did you put those bars exactly? A larger screenshot would help. :)

**Reply by PovilasID** [reply]
I added the bars the cells in the table at the top displaying averages
Here are a couple of screenshots: https://i.imgur.com/N9kwSpH.png https://i.imgur.com/9AxBFUn.png

**Comment by shalak** [comment]
Does the InfluxDB feature replace the default DB used by Speedtest Tracker, or is the InfluxDB just a sink, that may or may not work and won't interfere with regular Speedtest Tracker work?

**Reply by alexdelprete** [reply]
> Does the InfluxDB feature replace the default DB used by Speedtest Tracker, or is the InfluxDB just a sink, that may or may not work and won't interfere with regular Speedtest Tracker work?

InfluxDB is an added as an optional additional data exporting feature. When ST executes a test, the internal db is updated and if you've configured Influxdb, data is also written there.

**Comment by crypticviper** [comment]
@alexdelprete thanks for the dash and the documentation on it. I was able to get it to work, however, some of the "latest" tiles are empty. Is there anything else to be configured for this data?

![image](https://github.com/user-attachments/assets/38780f8e-8854-41e5-91a0-c7f253f0c728)
